### DISCUSSION TITLE
Question: How to use a Google Cast Custom Web Receiver with Mux Player

### DISCUSSION THREAD

**Discussion by sandervdv** [muxinc/elements#433] [Q&A]
Hi,

I managed to use a custom Google Cast Custom Web Receiver in combination with Mux Player in the following way:

		const muxPlayer = document.querySelector("mux-player"); 
		muxPlayer.requestCast({receiverApplicationId: 'XXXXXXX'});

Is it possible to provide a custom receiverApplicationId in Mux Player when using the default MediaCastButton? The above code is trigged by a custom button.

I also wonder how Mux Data is used in combination with Chrome Cast viewers, since the default receiver is used I assume Mux Data is not active within the Chrome Cast receiver? Are these viewers still tracked by Mux Data using Mux Player?

My end goal is to have a backup stream feature for when the primary stream fails, within Mux Player I can catch the ended event and switch playback-id. For Chrome Cast this will be much more difficult and I think I will need to build a custom receiver that can detect issues and switch to a backup stream. This will also require some custom exchange of data between the player and the receiver. Probably not easy to implement with Mux Player, as the extensibility seems to be limited. Any pointers would be helpful!

Thanks,
Sander

**Comment by luwes** [comment]
Hi @sandervdv,

Thanks for the feedback and glad you found the `requestCast()` method for a workaround.
It's currently not possible yet to set a custom receiver app ID but this was definitely in our plans, we will add this to the roadmap.

The best way to ensure Mux data is tracking everything and as a separate session is to install the Mux SDK in your custom receiver application. https://docs.mux.com/guides/data/monitor-chromecast

However Mux player on its own will continue to track video playback events that are send from the receiver to sender (Mux player).
Of course when the connection gets lost between sender and receiver, the player won't be able to track anymore.

Supporting custom Cast receivers with fallback mechanisms might be difficult at the moment with Mux player.
Although if I remember correctly custom messages between sender and receiver might not take that much code.
We built pretty much everything in the open so feel free to take a look at the code at https://github.com/muxinc/castable-video and we might be able to add in support for custom messages via a PR

**Comment by dylanjha** [comment]
I heard of someone else doing this by over-riding the global variable for `chrome.cast.media.DEFAULT_MEDIA_RECEIVER_APP_ID` and assigning it to their own app ID. Although slightly hacky that's another option until we have a direct way of specifying a custom receiver app
